scripts/rollup/build.js

```diff
@@ -564,7 +564,8 @@ async function createBundle(bundle, bundleType) {
   const rollupConfig = {
     input: resolvedEntry,
     treeshake: {
-      pureExternalModules,
+      moduleSideEffects: (id, external) =>
+        !(external && pureExternalModules.includes(id)),
```

Comment:
Refer to the rollup source code for how moduleSideEffects emulates the behavior of pureExternalModules, see https://github.com/rollup/rollup/blob/v1.19.4/src/ModuleLoader.ts#L88